maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/report/StatelessXmlReporter.java

```diff
@@ -451,7 +451,9 @@ private static void getTestProblems(
                     String type = delimiter == -1 ? stackTrace : stackTrace.substring(0, delimiter);
                     ppw.addAttribute("type", type);
                 } else {
-                    ppw.addAttribute("type", new StringTokenizer(stackTrace).nextToken());
+                    ppw.addAttribute(
+                            "type",
+                            isBlank(stackTrace) ? "UndefinedException" : new StringTokenizer(stackTrace).nextToken());
```

Comment:
Why don't you reuse the approach from line 451?

Reply:
Do you mean `indexOf` approach or `to use value as is`?

I didn't want to leave a chance for `type` to be null. What non-null value can be used here?

The `stackTrace` StatelessXmlReporter#439 var may be null:
```
public String getStackTrace(boolean trimStackTrace) {
        StackTraceWriter w = original.getStackTraceWriter();
        return w == null ? null : (trimStackTrace ? w.writeTrimmedTraceToString() : w.writeTraceToString());
    }
```
At the same time surefire's XSD requires `type`
<details>
 <summary>XSD</summary>

```xml
<xs:element name="error" nillable="true" minOccurs="0" maxOccurs="1">
                                <xs:complexType>
                                    <xs:simpleContent>
                                        <xs:extension base="xs:string">
                                            <xs:attribute name="message" type="xs:string"/>
                                            <xs:attribute name="type" type="xs:string" use="required"/>
                                        </xs:extension>
                                    </xs:simpleContent>
                                </xs:complexType>
                            </xs:element>
```
</details>

The goal is to handle stack traces like:
```
// has message
junit.framework.ComparisonFailure: 
Expected :fail at foo
Actual   :faild at foo

// does not have message
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:87)
	at org.junit.Assert.fail(Assert.java:96)

// or stack trace is missing (empty String)

```

Please, let me know if this alternative is more preferable:
<details>
 <summary>Option A</summary>

```java
if (report.getStackTraceWriter() != null) {
            //noinspection ThrowableResultOfMethodCallIgnored
            SafeThrowable t = report.getStackTraceWriter().getThrowable();
            if (t != null) {
                int delimiter = StringUtils.indexOfAny(stackTrace, ":", "\t", "\n", "\r", "\f");
                String type = delimiter == -1 ? stackTrace : stackTrace.substring(0, delimiter);
                ppw.addAttribute("type", Objects.toString(type, "UndefinedException"));
            }
        }
```
</details>
<details>
 <summary>Option B</summary>

```java
SafeThrowable t = report.getStackTraceWriter().getThrowable();
            if (t != null) {
                if (t.getMessage() != null) {
                    int delimiter = stackTrace.indexOf(":");
                    String type = delimiter == -1 ? stackTrace : stackTrace.substring(0, delimiter);
                    ppw.addAttribute("type", type);
                } else {
                    ppw.addAttribute(
                            "type",
                            isBlank(stackTrace) ? stackTrace : new StringTokenizer(stackTrace).nextToken());
                }
            }
```
</details>
<details>
 <summary>Option C</summary>

```java
SafeThrowable t = report.getStackTraceWriter().getThrowable();
            if (t != null) {
                if (t.getMessage() != null) {
                    int delimiter = stackTrace.indexOf(":");
                    String type = delimiter == -1 ? stackTrace : stackTrace.substring(0, delimiter);
                    ppw.addAttribute("type", type);
                } else {
                    int delimiter = StringUtils.indexOfAny(stackTrace, "\t", "\n", "\r", "\f");
                    String type = delimiter == -1 ? stackTrace : stackTrace.substring(0, delimiter);
                    ppw.addAttribute("type", type);
                }
            }
</details>

Reply:
I looked at this again and I want to address for 3.3.0.

I have option D for you: Since this is an oversight and `strackTrace` can be truly null of `t.getMessage()` is `null`, we should not write non-sense or empty to `type`, but leave it out. Rather no output than bad output. My proposal: Since we know that `type` can be null, I can remove the `requried` in the schema and bump to 3.1.0.

WDYT?

Reply:
Let me then prepare a PR with that change and you can build up on it.